### ISSUE TITLE
Battlefield Bad Company 2 is odd and confusing

### ISSUE THREAD

**Issue by dgibbs64** [gamedig/node-gamedig#514]
**Describe the bug**
The new name change for Battlefield Bad Company 2 from `bfbc2` to `bbc2` is odd and confusing.

I recommend either keeping the old name of bfbc2 which is the known shorthand name of the game (All Battlefield games shorthand are known as bf* for example bf2, bf3, bf2142 etc) or renaming it to something battlefieldbc2 might be more appropriate. I can see that new rules have been created for naming which is great. However in this case it doesn't seem to work and bending the rules might be a good idea in cases like this.

**Comment by cetteup** [comment]
I totally agree that names should be somewhat consistent within a game/franchise

Arma is another weird one as it stands. Just Arma: Armed Assault is "aaa". Arma 2 - Operation Arrowhead is "a2oa". Arma Reforger, however, is "armareforger". Meaning by looking for "arma" in the IDs, you'd find some Arma games - but not all. The sorting also does not help, since different games of a franchise will be sorted to totally different positions due to how different their names are.

I know incorporating franchises into the naming convention is difficult - especially since what is a single game now may become a franchise down the road. But maybe there's a compromise somewhere.

**Comment by podrivo** [comment]
You can still use old ids by adding the `checkOldIDs` when querying servers. More [here](https://github.com/gamedig/node-gamedig/blob/master/MIGRATE_IDS.md).

**Comment by dgibbs64** [comment]
The old id's will eventually go away so that is not really a solution. My suggestion (while v5 is still in test) is to review the game IDs to see if they make sense to humans and appear consistent. If its hard to guess what the game is just by looking at the gameid then there is a problem. Using the new rules as guidelines rather hard rules you must abide by. Otherwise you will end up with some gameids that make no sense to people. @cetteup gave some good examples of this issue.

**Comment by podrivo** [comment]
There was a PR (https://github.com/gamedig/node-gamedig/pull/487) to add game aliases for a more user friendly approach, but the discussion hasn't evolved, as we had to focus on other stuff at the moment.

**Comment by CosminPerRam** [comment]
We have decided that we should standardize GIDs, and while doing so we established ([rust-GameDig#108](https://github.com/gamedig/rust-gamedig/pull/108)) some rules of how they should be formed, as until gids refactor merged into node-gamedig (https://github.com/gamedig/node-gamedig/pull/415) they were decided pretty much at random and personal preference (by the person that made the PR), some games ending up with nice ids, some not and others with weird ones.

I'm thinking that gids shouldn't really matter, as if you were to select from a drop down list what game you want to query, you wouldn't let the user select from the ids, but from the names.

> The sorting also does not help, since different games of a franchise will be sorted to totally different positions due to how different their names are.

So would sorting by game name instead of id help here?

> However in this case it doesn't seem to work and bending the rules might be a good idea in cases like this.

We cant really 'bend the rules', as doing so for a game would be wanted to do so for another, and another, and so on. It would also break CI gids testing.

A middle ground to this (between strict gids and game names) would be something like @podrivo mentioned, adding another field that would represent a popular name for that game, and be able to query by that, or maybe directly by the name. thoughts?

Don't get me wrong, I'd love to have better readable gids, but the problem is that we really cannot create a set of rules that makes all game names into nicely ids.

**Comment by CosminPerRam** [comment]
@dgibbs64 @cetteup any opinions related to my last comment?

**Comment by MrWook** [comment]
@CosminPerRam I'm just curious—why did you decide to use abbreviations for game names instead of slugified game names? Using slugs would completely eliminate issues with rules, duplicates, and incorrect ordering. 
Sure some of them would be pretty long like `grand-theft-auto-san-andreas-multi-theft-auto` but does it matter?

**Comment by CosminPerRam** [comment]
> why did you decide to use abbreviations for game names?

The library originally used abbreviations, and the person that did the PR pretty much decided on their own on what the abbreviation would be, now we got a system for those that we can also test.

> instead of slugified game names?

I haven't thought of this, at a glance this looks like a solid option, I'll look into it, thanks.

> some of them would be pretty long [...] but does it matter?

No, I don't really think it does.

**Comment by podrivo** [comment]
I'd argue that a big name like that would be hard to use on a daily basis by users!

Sure, it's not that big of a deal if you set this once and let it run, or if we're doing our internal tests for development.
But for users, it's sure not good to use!

We can definitely have the official ID like `grand-theft-auto-san-andreas-multi-theft-auto`, but we should also consider adding an alias for easier use, like `gta-san-andreas-mta`. Same goes for the Battlefield Bad Company 2 ID.

My point is having something that considers more the user perspective!

**Comment by mmorrisontx** [comment]
I know I totally missed the boat on this, but it's very odd to totally break backward compatibility for something as trivial as an ID list. You can choose whatever you want as the primary id for standardization, but why not just support the old IDs forever? Disk space is cheap, engineering time is expensive.

**Comment by CosminPerRam** [comment]
Well said, considering that even when we changed from v4 to v5 we didnt had any ID collisions, as long as we dont have any after another potential change, we wouldn't have a reason to completely remove them (the migration document mentions that they'll be removed eventually but we dont have to).
